engine.py

```diff
@@ -379,6 +379,38 @@ def generate_thumbnail(self, document=None, output_path=None):
 
             return output_path
 
+    def save_as(self, document):
+        """
+        Launch a Qt file browser to select a file, then save the supplied
+        document to that path.
+
+        :param document: The document to save.
+        """
+
+        from sgtk.platform.qt import QtGui
+
+        try:
+            doc_path = document.fullName.fsName
```

Comment:
I can't use that method because I need the path for the supplied document, which isn't necessarily the active one.